twitterpuller/src/main/java/de/brandwatch/minianalytics/twitterpuller/TwitterPullerApplication.java

```diff
@@ -1,12 +1,26 @@
 package de.brandwatch.minianalytics.twitterpuller;
 
+import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
+import de.brandwatch.minianalytics.twitterpuller.twitter.TweetFetcher;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+import javax.annotation.PostConstruct;
+
 @SpringBootApplication
 public class TwitterPullerApplication {
 
     public static void main(String[] args) {
         SpringApplication.run(TwitterPullerApplication.class, args);
     }
+
+    @Autowired
+    Producer producer;
```

Comment:
can be private

as I made that comment a few times now, a general advise: Try to make things as restricted as you can, and as unrestricted as you have to. Meaning: Don't do public if you can do protected, and don't do protected if you can do private.

Same goes for immutability: Do final when you can. Do static when you can. All these things make you program either more secure, or more performant, or both.